stockflux-securities-master/src/components/form-view/form/Form.js

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import { AlertType } from '../../alerts/AlertType';
+import ToggleSwitch from '../form-controls/toggle-switch/ToggleSwitch';
+import TextField from '../form-controls/text-field/TextField';
+import * as action from '../../../actions/Security';
+import * as service from '../../../services/SecuritiesService';
+import securityReducer, { initialState } from '../../../reducers/Security';
+import './Form.css';
+
+const Form = ({ securityId, setAlerts, setRedirect }) => {
+  const [state, dispatch] = useReducer(securityReducer, initialState);
+
+  useEffect(() => {
+    if (securityId) {
+      service
+        .getSecurity(securityId)
+        .then(security => {
+          setSecurity(security);
+        })
+        .catch(() => {});
```

Comment:
What happens if the service fails to fetch a security? Are errors handled somewhere else maybe?

Reply:
if this is not handling anything then don't do .catch at all and let the exception bubble up. Please dont'consume and do nothing.

Reply:
Totally agree, thanks for spotting this!

Reply:
It might be worth adding some error handling instead of removing the catch here. I think it would be beneficial to display a message to the user if the service fails to fetch the security

Reply:
Yeah, actually, you're right. Just added it now.